src/main/sensors/barometer.c

```diff
@@ -322,28 +324,43 @@ bool isBaroReady(void) {
 
 uint32_t baroUpdate(void)
 {
-    static barometerState_e state = BAROMETER_NEEDS_SAMPLES;
+    static barometerState_e state = BAROMETER_NEEDS_PRES_START;
 
     switch (state) {
         default:
-        case BAROMETER_NEEDS_SAMPLES:
+        case BAROMETER_NEEDS_TEMP_START:
+            baro.dev.start_ut(&baro.dev);
+            state = BAROMETER_NEEDS_TEMP_SAMPLE;
+            return baro.dev.ut_delay;
+            break;
+
+        case BAROMETER_NEEDS_TEMP_SAMPLE:
             baro.dev.get_ut(&baro.dev);
+            state = BAROMETER_NEEDS_PRES_START;
+        break;
```

Comment:
The intention is to split the baro task into separately schedulable parts, but to execute them as quickly as possible. The behaviour is exactly as intended. A delay of  baro.dev.up_delay is NOT what's needed. The code now runs nearly as fast as it did without messing up the loop time.

Reply:
It will still negatively affect the gyro loop scheduling and delay it approximately every 23ms. As you have it now it will be:

start pressure ADC (33us)
delay 10us
read pressure/temp (93us)

Resulting in 136us. 8k gyro loop is 125us. The 10us delay between the tasks is not enough time to give the gyro task much opportunity for scheduling. We can't do anything about improving the gyro scheduling without slowing down the baro sampling.

Reply:
Let me hook up the logic analyser again and check the behaviour. I'm having trouble squashing at the moment. Let me fix that first.

Reply:
Any idea what the issue is with the following?

```
$ git rebase -i master
error: could not apply fa353f9... Split baro processing from two to four parts as reading the ADC and starting the next sample was taking too long

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".

Could not apply fa353f9cbacb6e1d6339034fa0649c2835803648... Split baro processing from two to four parts as reading the ADC and starting the next sample was taking too long
```

Reply:
Looks like a merge conflict - manually fix it, then do a `git add .; git rebase --continue`.